### PR TITLE
Move ToC to the beginning of the firmware image

### PR DESCRIPTION
**Pull request by pussuw**
Depends on: https://github.com/tiiuae/saluki-v1/pull/35

### CODE REVIEW

File: platforms/nuttx/src/bootloader/common/image_toc.c [hunk 1]
```diff
@@ -50,11 +50,16 @@
 #endif
 #define FLASH_END_ADDRESS (FLASH_START_ADDRESS + BOARD_FLASH_SIZE)
 
-bool find_toc(const image_toc_entry_t **toc_entries, uint8_t *len)
+/* If the ToC location is not explicitly stated, define a default value */
+#ifndef TOC_ADDRESS
+#define TOC_ADDRESS APP_LOAD_ADDRESS + BOOT_DELAY_ADDRESS + 8
```
Comment (pussuw) [inline platforms/nuttx/src/bootloader/common/image_toc.c:55]: Maybe this is not necessary anymore, if we can break backwards compatibility with firmwares that have ToC + signature.

Comment (jlaitine) [reply platforms/nuttx/src/bootloader/common/image_toc.c:55]: This is probably not needed. Instead, define this as an example in px4/fmu-v5/src/hw_config.h . everyone can just define where their TOC lies

**Review by jlaitine — APPROVED** [review]

**Comment by jlaitine** [comment]
LGTM; you can decide whether you want to have the TOC_ADDRESS defined there or not

**Review by jlaitine — APPROVED** [review]

**Comment by pussuw** [comment]
Pixhawk4/5 need to be fixed (CI error), I'll fix those

**Pull request merged**